src/main/haskell/kore/src/Kore/OnePath/Step.hs

```diff
@@ -298,16 +302,31 @@ transitionRule
             ExpandedPattern.simplify
                 tools substitutionSimplifier simplifier result
         let
+            unfilteredOrPatterns = ExpandedPattern.extractPatterns orResult
+            refuteIfPossible
+                expanded@Predicated {predicate=expandedPredicate}
+              = do
+                truthValue <- Predicate.refutePredicate expandedPredicate
+                case truthValue of
+                    Just True ->
+                        return expanded
+                            {Predicated.predicate = makeTruePredicate}
+                    Just False ->
+                        return ExpandedPattern.bottom
+                    Nothing -> return expanded
```

Comment:
Seems like there's some logical duplication with `Step.Condition.Evaluator` here. 

Perhaps we could refactor `OrOfExpandedPattern.isTrue` and `OrOfExpandedPattern.isFalse` to also return `Maybe Bool`, and then combine the partial information with `<>`